src/template/wasm2c.declarations.c

```diff
@@ -613,3 +619,10 @@ DEFINE_TABLE_FILL(externref)
 #define FUNC_TYPE_EXTERN_T(x) const char x[]
 #define FUNC_TYPE_T(x) static const char x[]
 #endif
+
+#if (__STDC_VERSION__ >= 201112L) || defined(_Static_assert)
+#define wasm_static_assert(X) _Static_assert(X, "assertion failure")
+#else
+#define wasm_static_assert(X) \
```

Comment:
I did a quick test, it looks like MSVC generates a warning for this while clang and gcc seem happy. Unfortunately, I can't test with mingw easily as it doesn't seem to be setup right on Godbolt. We should do a sanity check to make sure all standard C compilers are ok with this construction.

Reply:
Hmm, my only access to MSVC is via our CI, and it appeared to work there without complaint (https://github.com/WebAssembly/wabt/actions/runs/6152502422/job/16694736954#step:14:2320). What's the warning?

Reply:
I am seeing `<source>(9): warning C4116: unnamed type definition in parentheses`
Here is the godbolt link https://godbolt.org/z/ePe66x15

Reply:
Is the `wasm_` prefix the correct one to use here?  Why not `wasm_rt_`?

Reply:
I guess ny thinking was that these static asserts aren't part of the runtime ("_rt_") API, or even something that happens at runtime at all.

Reply:
Is `wasm_` the right prefix though?   What else do we put in `wasm_`?   There are not public symbol that exist outside the TU, right?  So perhaps simply `static_assert` with no namespace?

Reply:
Unfortunately `static_assert` is a reserved keyword in C23 and C++11. :-(

Reply:
Perhaps we can detect that and not define this at all in those cases?  i.e. define `static_assert` if and only if the compiler didn't already.